lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1625,6 +1625,8 @@ def check_gc(self, gc, fillcolor=None):
         gc._fillcolor = fillcolor
 
         orig_alphas = getattr(gc, '_effective_alphas', (1.0, 1.0))
+        if gc.get_rgb() is None:
+            gc.set_foreground([0.0, 0.0, 0.0, 1.0])
```

Comment:
I'm not entirely sure what the alpha value has to do. I simply revert the unwanted behavior of #5596 i.e. in the function Line2D.draw(renderer), with the above setting, prior to the the change-set, at line `830`, `edgecolor` would be `'k'`, and after the call `gc.set_foreground(edgecolor)` on line `835`, `gc._rgb` takes the value that I set. Will put a test case tomorrow to check to be sure that I did not miss anything.